src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs

```diff
@@ -40,31 +52,103 @@ pub struct NativeDeserializeDataTransform {
     output_data: Option<DataBlock>,
     parts: Vec<PartInfoPtr>,
     chunks: Vec<DataChunks>,
+
+    prewhere_columns: Vec<usize>,
+    remain_columns: Vec<usize>,
+
+    src_schema: DataSchema,
+    output_schema: DataSchema,
+    prewhere_filter: Arc<Option<Expr>>,
+
+    prewhere_skipped: usize,
 }
 
 impl NativeDeserializeDataTransform {
     pub fn create(
         ctx: Arc<dyn TableContext>,
         block_reader: Arc<BlockReader>,
+        plan: &DataSourcePlan,
```

Comment:
Can we remove the `plan` to use `pushdwon`?
We have tried to decouple the planner with storage.

Reply:
we also need the plan.source_info in `NativeDeserializeDataTransform`

Reply:
`plan.source_info.schema()` --> `schema`

Reply:
We have done much work to decouple `planners` for many crates in the past.

Reply:
Oh, i see.
This is common_catalog, not planners.